src/Utils/FileSystem.php

```diff
@@ -84,6 +84,14 @@ public static function delete($path)
 			if (!@rmdir($path)) { // @ is escalated to exception
 				throw new Nette\IOException("Unable to delete directory '$path'.");
 			}
+		} elseif ($path instanceof \Nette\Utils\Finder) {
+			/** @var \SplFileInfo $item */
+			foreach ($path as $item) {
+				static::delete($item->getRealPath());
+			}
+			if (!@rmdir($item->getPath())) { // @ is escalated to exception
```

Comment:
I wanted to delete all selected files and the directory in which they are. There is missing condition on empty directory which should be removed. And second problem is that files can be in different directories - I dont know if it is possible with Finder.